src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -444,6 +450,9 @@ private String generateQuery(TableReference tableRef) {
         case UPDATE:
           return String.format(UPDATE_QUERY, destinationTable, fieldsForUpdate, sourceTable, criteria);
         case UPSERT:
+          String orderedBy = orderedByList.isEmpty() ? "" : "ORDER BY " + String.join(", ", orderedByList);
+          sourceTable = String.format(SOURCE_DATA_QUERY, String.join(", ", tableKeyList), orderedBy,
```

Comment:
the same thing should be done for update if an order is specified. Otherwise I assume the existing merge query just applies it in whatever order it chooses?

Reply:
Updated. New logic applied to Update and Upsert operation.